integrationtest plugin causes build failure with version 0.11.2, works in 0.11.1

Hi!

I encountered an issue with pybuilder version 0.11.2. Running "pyb analyze" fails with

```
[INFO]  Executing flake8 on project sources.
[WARN]  flake8: src/integrationtest/python:1:1: E902 IOError: [Errno 2] No such file or directory: 'src/integrationtest/python'
------------------------------------------------------------
BUILD FAILED - flake8 found 1 warning(s)
------------------------------------------------------------
```
You can see this on Travis CI:
https://travis-ci.org/ImmobilienScout24/afp-core/builds/88392996
When I downgrade pybuilder to version 0.11.1, it works:
https://travis-ci.org/ImmobilienScout24/afp-core/builds/88395058

The reason is that my build.py includes the integrationtest plugin, but there is actually no /src/integrationtest/ directory. Version 0.11.1 did not have a problem with that. When I remove the plugin from my build.py, things work again, even with 0.11.2.

--- a/src/main/python/pybuilder/plugins/python/python_plugin_helper.py
+++ b/src/main/python/pybuilder/plugins/python/python_plugin_helper.py
@@ -59,13 +59,17 @@ def discover_affected_files(include_test_sources, include_scripts, project):
 def discover_affected_dirs(include_test_sources, include_scripts, project):
     files = [project.get_property("dir_source_main_python")]
     if include_test_sources:
-        if project.get_property("dir_source_unittest_python"):
+        if _if_property_set_and_dir_exists(project.get_property("dir_source_unittest_python")):
             files.append(project.get_property("dir_source_unittest_python"))
-        if project.get_property("dir_source_integrationtest_python"):
+        if _if_property_set_and_dir_exists(project.get_property("dir_source_integrationtest_python")):
             files.append(project.get_property("dir_source_integrationtest_python"))
-    if include_scripts and project.get_property("dir_source_main_scripts"):
+    if include_scripts and _if_property_set_and_dir_exists(project.get_property("dir_source_main_scripts")):
         files.append(project.get_property("dir_source_main_scripts"))
     return files
+
+
+def _if_property_set_and_dir_exists(property_value):
+    return property_value and os.path.isdir(property_value)
 
 
 def execute_tool_on_source_files(project, name, command_and_arguments, logger=None,

--- a/src/unittest/python/plugins/python/python_plugin_helper_tests.py
+++ b/src/unittest/python/plugins/python/python_plugin_helper_tests.py
@@ -23,7 +23,8 @@ from mock import Mock, call, patch
 from pybuilder.plugins.python.python_plugin_helper import (log_report,
                                                            discover_affected_files,
                                                            discover_affected_dirs,
-                                                           execute_tool_on_source_files)
+                                                           execute_tool_on_source_files,
+                                                           _if_property_set_and_dir_exists)
 
 
 class LogReportsTest(unittest.TestCase):
@@ -138,7 +139,8 @@ class DiscoverAffectedDirsTest(unittest.TestCase):
         files = discover_affected_dirs(False, False, project)
         self.assertEqual(files, ['source_directory'])
 
-    def test_should_discover_source_dirs_when_test_sources_are_included(self):
+    @patch('pybuilder.plugins.python.python_plugin_helper.os.path.isdir', return_value=True)
+    def test_should_discover_source_dirs_when_test_sources_are_included(self, _):
         project = Mock()
 
         project.get_property.side_effect = lambda _property: _property
@@ -154,7 +156,8 @@ class DiscoverAffectedDirsTest(unittest.TestCase):
         self.assertEquals(files,
                           ['dir_source_main_python', 'dir_source_unittest_python', 'dir_source_integrationtest_python'])
 
-    def test_should_discover_source_dirs_when_test_sources_are_included_no_unittests(self):
+    @patch('pybuilder.plugins.python.python_plugin_helper.os.path.isdir', return_value=True)
+    def test_should_discover_source_dirs_when_test_sources_are_included_no_unittests(self, _):
         project = Mock()
 
         project.get_property.side_effect = lambda \
@@ -170,7 +173,8 @@ class DiscoverAffectedDirsTest(unittest.TestCase):
         self.assertEquals(files,
                           ['dir_source_main_python', 'dir_source_integrationtest_python'])
 
-    def test_should_discover_source_dirs_when_test_sources_are_included_no_integrationtests(self):
+    @patch('pybuilder.plugins.python.python_plugin_helper.os.path.isdir', return_value=True)
+    def test_should_discover_source_dirs_when_test_sources_are_included_no_integrationtests(self, _):
         project = Mock()
 
         project.get_property.side_effect = lambda \
@@ -186,7 +190,8 @@ class DiscoverAffectedDirsTest(unittest.TestCase):
         self.assertEquals(files,
                           ['dir_source_main_python', 'dir_source_unittest_python'])
 
-    def test_should_discover_source_dirs_when_script_sources_are_included(self):
+    @patch('pybuilder.plugins.python.python_plugin_helper.os.path.isdir', return_value=True)
+    def test_should_discover_source_dirs_when_script_sources_are_included(self, _):
         project = Mock()
 
         project.get_property.side_effect = lambda \
@@ -200,6 +205,21 @@ class DiscoverAffectedDirsTest(unittest.TestCase):
                           call('dir_source_main_scripts')])
         self.assertEquals(files,
                           ['dir_source_main_python', 'dir_source_main_scripts'])
+
+    @patch('pybuilder.plugins.python.python_plugin_helper.os.path.isdir', return_value=True)
+    def test_if_property_set_and_dir_exists(self, exists):
+        self.assertTrue(_if_property_set_and_dir_exists('test'))
+        exists.assert_called_once_with('test')
+
+    @patch('pybuilder.plugins.python.python_plugin_helper.os.path.isdir', return_value=False)
+    def test_if_property_set_and_dir_not_exists(self, exists):
+        self.assertFalse(_if_property_set_and_dir_exists('test'))
+        exists.assert_called_once_with('test')
+
+    @patch('pybuilder.plugins.python.python_plugin_helper.os.path.isdir', return_value=False)
+    def test_if_property_not_set_and_dir_not_exists(self, exists):
+        self.assertFalse(_if_property_set_and_dir_exists(None))
+        exists.assert_not_called()
 
 
 class ExecuteToolOnSourceFilesTest(unittest.TestCase):

